Source/WebKit/WebProcess/WebPage/Cocoa/TextAnimationController.mm

```diff
@@ -113,49 +112,43 @@
         return std::nullopt;
     }
 
-    return corePage->contextRangeForSessionWithID(sessionID);
+    return corePage->contextRangeForActiveWritingToolsSession();
 }
 
 std::optional<WebCore::SimpleRange> TextAnimationController::contextRangeForTextAnimationID(const WTF::UUID& animationUUID) const
 {
-    if (auto iterator = m_manuallyEnabledAnimationRanges.find(animationUUID); iterator != m_manuallyEnabledAnimationRanges.end())
-        return WebCore::makeSimpleRange(iterator->value.get());
+    if (RefPtr manuallyEnabledAnimationRange = m_manuallyEnabledAnimationRange; manuallyEnabledAnimationRange)
+        return WebCore::makeSimpleRange(*manuallyEnabledAnimationRange);
 
-    for (auto [sessionID, initialAnimationID] : m_initialAnimations) {
-        if (initialAnimationID == animationUUID)
-            return unreplacedRangeForSessionWithID(sessionID);
-    }
+    if (m_initialAnimationID == animationUUID)
+        return unreplacedRangeForActiveWritingToolsSession();
 
-    for (auto [sessionID, animationRanges] : m_textAnimationRanges) {
-        for (auto animationRange : animationRanges) {
-            if (animationRange.animationUUID == animationUUID) {
-                if (auto fullSessionRange = contextRangeForSessionWithID(sessionID))
-                    return WebCore::resolveCharacterRange(*fullSessionRange, animationRange.range, defaultTextAnimationControllerTextIteratorBehaviors);
-            }
+    for (auto animationRange : m_textAnimationRanges) {
+        if (animationRange.animationUUID == animationUUID) {
+            if (auto fullSessionRange = contextRangeForActiveWritingToolsSession())
+                return WebCore::resolveCharacterRange(*fullSessionRange, animationRange.range, defaultTextAnimationControllerTextIteratorBehaviors);
         }
     }
 
-    for (auto unstyledRangeData : m_unstyledRanges.values()) {
-        if (unstyledRangeData->animationUUID == animationUUID)
-            return unstyledRangeData->range;
-    }
+    if (m_unstyledRange && m_unstyledRange->animationUUID == animationUUID)
+        return m_unstyledRange->range;
 
     return std::nullopt;
 }
 
-void TextAnimationController::removeTransparentMarkersForSessionID(const WebCore::WritingTools::Session::ID& sessionID)
+void TextAnimationController::removeTransparentMarkersForActiveWritingToolsSession()
 {
-    if (auto iterator = m_initialAnimations.find(sessionID); iterator != m_initialAnimations.end())
-        removeTransparentMarkersForTextAnimationID(iterator->value);
+    if (auto initialAnimationID = m_initialAnimationID)
+        removeTransparentMarkersForTextAnimationID(*initialAnimationID);
 
-    auto animationStates = m_textAnimationRanges.take(sessionID);
+    auto animationStates = m_textAnimationRanges;
```

Comment:
this should set m_textAnimationRanges to { } i think

Reply:
I think yes, that you can clear/reset basically everything in the function. Perhaps it should be renamed as such.